src/QsCompiler/CompilationManager/CompilationUnitManager.cs

```diff
@@ -37,6 +38,11 @@ public class CompilationUnitManager : IDisposable
         /// </summary>
         private readonly CompilationUnit compilationUnit;
 
+        /// <summary>
+        /// Logs messages that occur during processing.
+        /// </summary>
+        private readonly Action<string, MessageType> logMessage;
```

Comment:
Because the exceptions can still sometimes be thrown as the result of an actual bug, logging them can be a hint to the developer if a language server feature is not working as expected. It is less confusing to have them be logged as "info" events than as errors.

Reply:
Ok, I see that logging is only in debug builds. More extensive documentation for the growing list of constructor arguments would be good. Could you please add that to the doc comment, and could you clarify the intended use of logMessage in the doc comment here more? What is the intended use exactly?

Reply:
I think it would actually be better to avoid adding another parameter to the constructor. I would like to either replace `Action<Exception> exceptionLogger` with `Action<string, MessageType> messageLogger`, or to replace both of them with a single logger object that can do both.

The logging situation is sort of confusing, though. There's an `ILogger` interface but it's defined in Compiler, so CompilationManager can't use it. I don't know if it's worth preserving the exception object the whole way through the logging process, or if converting it to a string that can be given to `Action<string, MessageType>` is fine. Do you have any thoughts on this?

Reply:
It is worth letting the language server decide what to do with an exception rather than converting it to a string before the server gets it. That would be important for a decent log and handling of these issues. You artificially want to hack how an exception looks like in a specific case, and I don't quite see the benefit of presenting it as something it isn't in general. Why not have the exception be what it is and have it be handled as such?

Reply:
There's nothing artificial about it. These two exceptions are special, and need to be handled differently from the other exceptions. That is the point of this PR.

Reply:
That gives me an idea, though. I think CompilationManager could handle them the same as other exceptions, but the language server can check the type of the exception object in its exception handler. That would mean that those two exceptions would be treated as errors when not running the language server (as they should be, since timing issues during updates shouldn't happen from `qsc`), but could be treated as less serious by the language server itself.

Reply:
I removed the new logger parameter.

Reply:
I like that much better! Thanks!